pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -1456,43 +1456,25 @@ func TestPodEligibleToPreemptOthers(t *testing.T) {
 			expected:            false,
 		},
 		{
-			name: "victim Pods terminating, feature PodDisruptionConditions is enabled",
-			fts:  feature.Features{EnablePodDisruptionConditions: true},
+			name: "victim Pods terminating",
 			pod:  st.MakePod().Name("p_with_nominated_node").UID("p").Priority(highPriority).NominatedNodeName("node1").Obj(),
 			pods: []*v1.Pod{st.MakePod().Name("p1").UID("p1").Priority(lowPriority).Node("node1").Terminating().
 				Condition(v1.DisruptionTarget, v1.ConditionTrue, v1.PodReasonPreemptionByScheduler).Obj()},
 			nodes:    []string{"node1"},
 			expected: false,
 		},
 		{
-			name:     "non-victim Pods terminating, feature PodDisruptionConditions is enabled",
-			fts:      feature.Features{EnablePodDisruptionConditions: true},
+			name:     "non-victim Pods terminating",
 			pod:      st.MakePod().Name("p_with_nominated_node").UID("p").Priority(highPriority).NominatedNodeName("node1").Obj(),
 			pods:     []*v1.Pod{st.MakePod().Name("p1").UID("p1").Priority(lowPriority).Node("node1").Terminating().Obj()},
 			nodes:    []string{"node1"},
 			expected: true,
 		},
-		{
-			name: "victim Pods terminating, feature PodDisruptionConditions is disabled",
-			fts:  feature.Features{EnablePodDisruptionConditions: false},
-			pod:  st.MakePod().Name("p_with_nominated_node").UID("p").Priority(highPriority).NominatedNodeName("node1").Obj(),
-			pods: []*v1.Pod{st.MakePod().Name("p1").UID("p1").Priority(lowPriority).Node("node1").Terminating().
-				Condition(v1.DisruptionTarget, v1.ConditionTrue, v1.PodReasonPreemptionByScheduler).Obj()},
-			nodes:    []string{"node1"},
-			expected: false,
-		},
-		{
-			name:     "non-victim Pods terminating, feature PodDisruptionConditions is disabled",
-			fts:      feature.Features{EnablePodDisruptionConditions: false},
-			pod:      st.MakePod().Name("p_with_nominated_node").UID("p").Priority(highPriority).NominatedNodeName("node1").Obj(),
-			pods:     []*v1.Pod{st.MakePod().Name("p1").UID("p1").Priority(lowPriority).Node("node1").Terminating().Obj()},
-			nodes:    []string{"node1"},
-			expected: false,
-		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			test.fts.EnablePodDisruptionConditions = true
```

Comment:
we generally remove this kind of code when graduating. Why are you keeping it?

Reply:
I didn't want to update the business logic in this PR (just tests), and this helper field is used in the logic [here](https://github.com/kubernetes/kubernetes/blob/5e331569e4cd461baf19b9f2cc493af1f7f7a73f/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L256). And as a boolean field it is defaulted to false, rather than to the actual feature gate state. Do you know why we use this indirection layer for feature gates in scheduler? Maybe we could simplify the code and use directly the feature gates, which would eliminate the confusion here.

Reply:
At some point there was an effort to make the scheduler framework code to be easily importable by other projects, which meant cutting dependencies.
This was not completed.